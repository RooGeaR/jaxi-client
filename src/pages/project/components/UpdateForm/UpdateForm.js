import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import { updateProject } from '../../../../mutations';
import { projectInfoQuery } from '../../../../queries';
import MutationForm from '../MutationForm';

const UpdateForm = ({ match: { params }, history }) => {

    return (
        <Query query={ projectInfoQuery } variables={ {id: params.projectId} }>
          {({ data, loading, error, fetchMore, refetch }) => {
            if (loading) return <p>loading...</p>;
            if (error) return <p>{error.message}</p>

            const projectValues = {
              title: data.project.title,
              user_id: data.project.user.id
            };

            return (
              <div className='row'>
                <div className={'col-md-5 col-md-offset-3'}>

                  <MutationForm 
                    action={ updateProject } 
                    isNewRecord={ false }
                    projectId={ Number(params.projectId) }
                    initialValues={ projectValues }
                    history={ history }
                  />
                </div>
              </div>

            );
          }}
        </Query>
    );
}

UpdateForm.propTypes = {
    /** Router params */
    match: PropTypes.object,

    /** Router history */
    history: PropTypes.object
}

export default UpdateForm;