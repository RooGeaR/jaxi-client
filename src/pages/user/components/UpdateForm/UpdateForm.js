import React from 'react';
import PropTypes from 'prop-types';
import { Query } from 'react-apollo';
import { updateUser } from '../../../../mutations';
import { userInfoQuery } from '../../../../queries';
import MutationForm from '../MutationForm';

const UpdateForm = ({ match: { params }, history }) => {

    return (
        <Query query={ userInfoQuery } variables={ {id: params.userId} }>
          {({ data, loading, error, fetchMore, refetch }) => {
            if (loading) return <p>loading...</p>;
            if (error) return <p>{error.message}</p>

            return (
              <div className='row'>
                <div className={'col-md-5 col-md-offset-3'}>

                  <MutationForm 
                    action={ updateUser } 
                    isNewRecord={ false }
                    userId={ Number(params.userId) }
                    initialValues={ data.user }
                    history={ history }
                  />
                </div>
              </div>

            );
          }}
        </Query>
    );
}

UpdateForm.propTypes = {
    /** Router params */
    match: PropTypes.object,

    /** Router history */
    history: PropTypes.object
}

export default UpdateForm;