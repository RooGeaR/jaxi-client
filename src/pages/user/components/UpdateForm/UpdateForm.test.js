import React from 'react';
import { shallow, render } from 'enzyme';
import UpdateForm from './UpdateForm';
import ApolloClient from 'apollo-boost';
import { ApolloProvider } from 'react-apollo';

const client = new ApolloClient({
    uri: "http://localhost:8088/graphql"
});

const routerMatch = {
  params: {
    userId: '1'
  }
};

describe('Update form', () => {
  it('should render correctly', () => {
    const component = shallow(<UpdateForm match={routerMatch} />);
  
    expect(component).toMatchSnapshot();
  });
});

describe('Update form', () => {
  it('should be render without error', () => {
    render(
      <ApolloProvider client={client}>
        <UpdateForm match={routerMatch} />
      </ApolloProvider>
    );
    
  });
});
