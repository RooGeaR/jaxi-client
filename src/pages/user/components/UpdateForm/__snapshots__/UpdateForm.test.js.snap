// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Update form should render correctly 1`] = `
<Query
  query={
    Object {
      "definitions": Array [
        Object {
          "directives": Array [],
          "kind": "OperationDefinition",
          "name": Object {
            "kind": "Name",
            "value": "getuser",
          },
          "operation": "query",
          "selectionSet": Object {
            "kind": "SelectionSet",
            "selections": Array [
              Object {
                "alias": undefined,
                "arguments": Array [
                  Object {
                    "kind": "Argument",
                    "name": Object {
                      "kind": "Name",
                      "value": "id",
                    },
                    "value": Object {
                      "kind": "Variable",
                      "name": Object {
                        "kind": "Name",
                        "value": "id",
                      },
                    },
                  },
                ],
                "directives": Array [],
                "kind": "Field",
                "name": Object {
                  "kind": "Name",
                  "value": "user",
                },
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "first_name",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "last_name",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "username",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "email",
                      },
                      "selectionSet": undefined,
                    },
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "projects",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "id",
                            },
                            "selectionSet": undefined,
                          },
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "title",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
              },
            ],
          },
          "variableDefinitions": Array [
            Object {
              "defaultValue": undefined,
              "directives": Array [],
              "kind": "VariableDefinition",
              "type": Object {
                "kind": "NonNullType",
                "type": Object {
                  "kind": "NamedType",
                  "name": Object {
                    "kind": "Name",
                    "value": "ID",
                  },
                },
              },
              "variable": Object {
                "kind": "Variable",
                "name": Object {
                  "kind": "Name",
                  "value": "id",
                },
              },
            },
          ],
        },
      ],
      "kind": "Document",
      "loc": Object {
        "end": 229,
        "start": 0,
      },
    }
  }
  variables={
    Object {
      "id": "1",
    }
  }
>
  <Component />
</Query>
`;
